testing/integration/headless/test_cdn.py

```diff
@@ -0,0 +1,250 @@
+        assert is_cdn_error(response)
+    else:
+        assert is_cdn_cache_miss(response)
+    return response
+
+
+def assert_not_cached(
+    url,
+    expected_status_code=200,
+    expected_location=None,
+    is_behind_cdn=True,
+    method="get",
+    **request_kwargs,
+):
+    if is_behind_cdn:
+        response1 = assert_not_cached_by_cdn(
+            url, expected_status_code, expected_location, method, **request_kwargs
+        )
+        response2 = assert_not_cached_by_cdn(
+            url, expected_status_code, expected_location, method, **request_kwargs
+        )
+        if expected_status_code in (301, 302) and not expected_location:
+            if "/users/github/login/" not in url:
+                assert response2.headers["location"] == response1.headers["location"]
+        return response2
+
+    response = request(method, url, **request_kwargs)
+    assert response.status_code == expected_status_code
+    assert "no-cache" in response.headers["Cache-Control"]
+    assert "no-store" in response.headers["Cache-Control"]
+    assert "must-revalidate" in response.headers["Cache-Control"]
+    assert "max-age=0" in response.headers["Cache-Control"]
+    return response
+
+
+def assert_cached(
+    url,
+    expected_status_code=200,
+    expected_location=None,
+    is_behind_cdn=True,
+    method="get",
+    **request_kwargs,
+):
+    response = request(method, url, **request_kwargs)
+    assert response.status_code == expected_status_code
+    if expected_status_code in (301, 302) and expected_location:
+        if callable(expected_location):
+            assert expected_location(response.headers["location"])
+        else:
+            assert response.headers["location"] == expected_location
+    if is_behind_cdn:
+        if is_cdn_cache_miss(response):
+            response2 = request(method, url, **request_kwargs)
+            assert response2.status_code == expected_status_code
+            assert is_cdn_cache_hit(response2)
+            if expected_status_code == 200:
+                assert response2.content == response.content
+            elif expected_status_code in (301, 302):
+                if expected_location:
+                    if callable(expected_location):
+                        assert expected_location(response2.headers["location"])
+                    else:
+                        assert response2.headers["location"] == expected_location
+                else:
+                    assert response2.headers["location"] == response.headers["location"]
+        else:
+            assert is_cdn_cache_hit(response)
+    else:
+        assert "public" in response.headers["Cache-Control"]
+        assert ("max-age" in response.headers["Cache-Control"]) or (
+            "s-maxage" in response.headers["Cache-Control"]
+        )
+    return response
+
+
+@pytest.mark.parametrize(
+    "slug,status,expected_location",
+    [
+        ("/miel", 500, None),
+        ("/_kuma_status.json", 200, None),
```

Comment:
I'll add those, thanks! The https://developer.mozilla.org/_kuma_status.json endpoint provides metadata about the deployment. I think the only place we use it currently is for our integration tests. Once we launch Yari, much of it might not be useful.